packages/rollup-full-node/src/app/web3-rpc-handler.ts

```diff
@@ -423,6 +427,35 @@ export class DefaultWeb3Handler
     return res
   }
 
+  public async getTransactionByHash(ovmTxHash: string): Promise<any> {
+    log.debug('Getting tx for ovm tx hash:', ovmTxHash)
+    // First convert our ovmTxHash into an internalTxHash
+    const internalTxHash = await this.getInternalTxHash(ovmTxHash)
+
+    if (!internalTxHash) {
+      log.debug(
+        `There is no EVM tx hash associated with OVM tx hash [${ovmTxHash}]`
+      )
+      return null
+    }
+
+    log.debug(
+      `OVM tx hash [${ovmTxHash}] is associated with EVM tx hash [${internalTxHash}]`
+    )
+
+    const internalTx = await this.context.provider.send(
+      Web3RpcMethods.getTransactionByHash,
+      [internalTxHash]
+    )
+
+    log.debug(
+      `OVM tx hash [${ovmTxHash}] is for EVM tx: [${JSON.stringify(
+        internalTx
+      )}]`
+    )
+    return internalTx
```

Comment:
Unfortunately my guess is that just returning the internal tx is probably not going to be sufficient for many use cases of this endpoint.  For example, if a dev 1. manually constructs an OVM transaction, only storing the hash 2. sends it off to the chain, and then 3. attempts to retrieve it later with this endpoint, it will not get back the original transaction constructed in step 1.